modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,616 @@
+                                LinkToParentHelper(nh, nodes);
+                                m_processed_nodes.insert(nh);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // 3. Link internal nodes to their parents
+        for (const auto& nh : nodes) {
+            if (m_gm.metadata(nh).get<NodeType>().t == NodeType::OP) {
+                GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                if (m_processed_nodes.find(nh) == m_processed_nodes.end()) {
+                    LinkToParentHelper(nh, nodes);
+                    m_processed_nodes.insert(nh);
+                }
+            }
+        }
+
+        m_device = std::unique_ptr<dai::Device>(new dai::Device(*m_pipeline));
+
+        // Prepare all output queues
+        for (size_t i = 0; i < outs_data.size(); ++i) {
+            // FIXME: add queue parameters
+            m_out_queues.push_back(m_device->getOutputQueue(m_out_queue_names[i], 30, true));
+        }
+    }
+
+void cv::gimpl::GOAKExecutable::handleNewStream() {
+    // do nothing
+}
+
+void cv::gimpl::GOAKExecutable::handleStopStream() {
+    // do nothing
+}
+
+void cv::gimpl::GOAKExecutable::run(GIslandExecutable::IInput  &in,
+                                    GIslandExecutable::IOutput &out) {
+    const auto in_msg = in.get();
+
+    if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg)) {
+        out.post(cv::gimpl::EndOfStream{});
+        return;
+    }
+
+    for (size_t i = 0; i < m_in_queues.size(); ++i) {
+        auto q = m_device->getInputQueue(m_in_queues[i].first);
+        q->send(m_in_queues[i].second);
+    }
+
+    for (size_t i = 0; i < m_out_queues.size(); ++i) {
+        auto q = m_out_queues[i];
+        auto oak_frame = q->get<dai::ImgFrame>();
+
+        auto out_arg = out.get(i);
+
+        if (util::holds_alternative<cv::detail::VectorRef>(out_arg)) {
+            cv::util::get<cv::detail::VectorRef>(out_arg).wref<uint8_t>() = oak_frame->getData();
+            // FIXME: should we copy data instead?
+        } else if (util::holds_alternative<cv::MediaFrame*>(out_arg)) {
+            // FIXME: hard-coded NV12
+            *cv::util::get<cv::MediaFrame*>(out_arg) =
+                    cv::MediaFrame::Create<cv::gapi::oak::OAKMediaAdapter>(
+                            cv::Size(static_cast<int>(oak_frame->getWidth()),
+                                     static_cast<int>(oak_frame->getHeight())),
+                            cv::gapi::oak::OAKFrameFormat::NV12,
+                            oak_frame->getData().data(),
+                            oak_frame->getData().data() + static_cast<long>(oak_frame->getData().size() / 3 * 2));
+            // FIXME: should we copy data instead?
+        } else {
+            GAPI_Assert(false && "Unsupported output type");
+        }
+
+        out.meta(out_arg, cv::util::get<cv::GRunArgs>(in_msg)[i].meta);
```

Comment:
please make sure it is filled up properly. I don't see any special handling here for the meta, so likely what you post is taken from the dummy stream source (may be incorrect).

The camera-defined values should be posted as `meta` here (see mx plugin for the reference).

Reply:
Good point, done